src/main/java/Storage.java

```diff
@@ -0,0 +1,82 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private File diskList;
+
+    public Storage(String filePath) {
+        diskList = new File(filePath);
+    }
+
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> taskList = new ArrayList<>();
+        try {
+            Scanner s = new Scanner(diskList);
+            while(s.hasNext()) {
+                String[] savedTask = s.nextLine().split(" \\| ");
+
+                switch (savedTask[0]) {
+                case "T":
+                    Todo t = new Todo(savedTask[2]);
+                    if (savedTask[1].equals("1")) {
+                        t.markAsDone();
+                    }
+                    taskList.add(t);
+                    break;
```

Comment:
Perhaps marking the task as done and adding it to the task list could be extracted out, instead of doing it over and over across each `case` statement (one for todo, one for deadline, and one for event)?